pkg/tsdb/loki/loki.go

```diff
@@ -166,12 +166,25 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 
 // we extracted this part of the functionality to make it easy to unit-test it
 func runQuery(ctx context.Context, api *LokiAPI, query *lokiQuery) (data.Frames, error) {
-	value, err := api.DataQuery(ctx, *query)
+	response, err := api.DataQuery(ctx, *query)
 	if err != nil {
 		return data.Frames{}, err
 	}
 
-	return parseResponse(value, query)
+	if response.Error != nil {
+		return data.Frames{}, err
```

Comment:
`api.DataQuery` returns a `DataResponse` and an `error`, that `error` is checked on line 170.
(in our case this covers error happening with the http-request going to Loki, or things like that)

`DataResponse` contains `Frames` and `Error`, that `Error` is checked on line 174.
(in our case this covers errors "created" by `converter/prom.go`)

Reply:
.. maybe we could simplify the code if we would modify `converter/prom.go` to return `([]data.Frame, error)` instead of `*DataResponse`, but i wanted to keep changes to `converter/prom.go` to a minimum here.

Reply:
@gabor i'm referring to line 175 where the `err` is returned. it seems like it would be nil based on the check on line 170, but i am probably missing something

Reply:
ok, fixed it i think.